### PR TITLE
Cluster wizard

### PR DESCRIPTION
**Pull request by qbalukom**
Reimplement adding a cluster as a wizard

### CODE REVIEW

File: core/src/i18n/en.yaml [hunk 1]
```diff
@@ -189,11 +189,15 @@ clusters:
   buttons:
     apply-configuration: Apply configuration
     apply-kubeconfig: Apply kubeconfig
+    verify-and-add: Verify & add
```
Comment (NHingerl) [inline core/src/i18n/en.yaml:192]:
```suggestion
    verify-and-add: Verify and Add
```
Formally, we need [title case for buttons](https://help.sap.com/viewer/DRAFT/e33c591ae4494a659a3f5f983c9d1161/PROD/en-US/79ce282052d14cc38e6a203a1fc277c4.html?q=buttons). Also we can't use [ampersands](https://help.sap.com/viewer/DRAFT/e33c591ae4494a659a3f5f983c9d1161/PROD/en-US/0511266eabeb4067b395b55e41280f26.html?q=ampersand) here.

Contentwise, do we really need to mention `verify` on the button or is this just a sub-step in order to `add`?  Meaning, why not simply call the button `Add`?

File: core/src/i18n/en.yaml [hunk 2]
```diff
@@ -226,6 +230,12 @@ clusters:
       sessionStorage: sessionStorage
       unknown: Unknown
     title: Storage Type
+  wizard:
+    kubeconfig: Choose configuration
+    update: Update configuration
+    storage: Storage mode
+    info: We don't store the kubeconfig file on server, it remains in the browser.
```
Comment (NHingerl) [inline core/src/i18n/en.yaml:237]:
```suggestion
    info: We don't store the kubeconfig file on the server, it remains in the browser.
```

File: core/src/i18n/en.yaml [hunk 3]
```diff
@@ -226,6 +230,12 @@ clusters:
...
+    incomplete: It looks like your kubeconfig is incomplete. Please fill the additional fields.
```
Comment (NHingerl) [inline core/src/i18n/en.yaml:238]:
```suggestion
    incomplete: It looks like your kubeconfig is incomplete. Fill out the additional fields.
```
No "please" needed :)

File: core/src/i18n/en.yaml (hunk 1 above)
Comment (qbalukom) [reply core/src/i18n/en.yaml:192]: To be honest, I was simply using what was given on designs. Just "Add" is fine with me

File: core/src/i18n/en.yaml [hunk 4]
```diff
@@ -222,10 +231,27 @@ clusters:
...
+    kubeconfig: Provide Kubeconfig
+    update: Provide Authentication
+    storage: Privacy Options
+    intro: To add a cluster, all you have to do is provide the cluster configuration – called “kubeconfig” in the Kubernetes world.
+    storage-info: |
+      Don't worry, we don't store your kubeconfig file on the server. The information stays in your browser.
+      You can choose your privacy options in the last step.
+    incomplete: We couldn't find enough authentication information in your kubeconfig file. You can enter it manually.
+    editor-label: 'or paste your kubeconfig here:'
+    token-info: Consult your authorization provider for information how to aquire the token.
```
Comment (NHingerl) [inline core/src/i18n/en.yaml:249]:
```suggestion
    token-info: If you don't know how to get your token, ask your authentication provider.
```

I believe it's "authentication" not "authorization" here, but correct me if I'm mistaken :)

File: core/src/i18n/en.yaml [hunk 5]
```diff
@@ -222,10 +231,27 @@ clusters:
...
+  wizard:
+    context: Current context
+    kubeconfig: Provide Kubeconfig
+    update: Provide Authentication
+    storage: Privacy Options
+    intro: To add a cluster, all you have to do is provide the cluster configuration – called “kubeconfig” in the Kubernetes world.
+    storage-info: |
+      Don't worry, we don't store your kubeconfig file on the server. The information stays in your browser.
+      You can choose your privacy options in the last step.
+    incomplete: We couldn't find enough authentication information in your kubeconfig file. You can enter it manually.
```
Comment (NHingerl) [inline core/src/i18n/en.yaml:247]:
```suggestion
    incomplete: We couldn't find enough authentication information in your kubeconfig. You can enter it manually.
```
Is it a file if the user only pasted the config?

File: core/src/i18n/en.yaml (hunk 4 above)
Comment (qbalukom) [reply core/src/i18n/en.yaml]: you're absolutely right

**Review by NHingerl — APPROVED** [review]

File: core-ui/src/components/Clusters/views/AddCluster/AddCluster.js [hunk 6]
```diff
@@ -1,49 +1,37 @@
...
-
-  React.useEffect(() => {
+  useEffect(() => {
     if (!isMounted.current) return; // avoid state updates on onmounted component
     if (!encodedParams || initParams) return;
     async function setKubeconfigIfPresentInParams() {
       const params = await decompressParams(encodedParams);
       setInitParams(params);
       if (Object.keys(params.kubeconfig || {}).length) {
+        console.log('set kubeconfig on params');
```
Comment (akucharska) [inline core-ui/src/components/Clusters/views/AddCluster/AddCluster.js:34]: This should probably be removed

**Review by akucharska — APPROVED** [review]

**Review by klaudiagrz — APPROVED** [review]

**Pull request merged**